packages/schematics/integration/nixt-facade/execute.ts

```diff
@@ -0,0 +1,22 @@
+const nixt = require('nixt');
+
+/**
+ * Execute the given commands using nixt#exec.
+ * @param commands Shell commands to run. Can be separated by &&, || or ;
+ * @param folder Working directory to execute the commands in. Default: .
+ */
+export function execute(command: string, folder = '.'): Promise<void> {
+  console.log(`Executing command ${command} in folder ${folder}`);
+
+  return new Promise((resolve, reject) => {
+    try {
+      nixt()
+        .cwd(folder)
+        .exec(command)
+        .run('', resolve);
```

Comment:
I need some help to understand this part.
I assume that both `exec` and `run` are synchronous operations.

Will `exec` provide the `stdout/stderr` that can be asserted later on?

Is it needed to write `run('', resolve)`?
Could we use `end` (https://github.com/vesln/nixt#end)?

Reply:
> Is it needed to write run('', resolve)?
Could we use end (https://github.com/vesln/nixt#end)?

```
.run('xyz', resolve);
```
 is the same as
```
.run('xyz')
.end(resolve);
```
I can change to the longer version, of you like.
---

> I assume that both exec and run are synchronous operations.
> Will exec provide the stdout/stderr that can be asserted later on?

Both are asynchronous. `exec` uses [child_process.exec](https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback) internally, and `run` uses [spawn](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options). That's why you can use command sequences (`command 1; command 2; ...`) only in exec. 

Nixt always expects a `run` call, though, and provides the `exec`-function more like a preparation tool for the the test setup. Also, only `run` provides the ability to create assertions. I'm actually not very satisfied by this API decision... maybe we should switch to another library or create one by our own?

Reply:
Thx, now I know what is going on.
I agree the API is not intuitive to me, either.

But let's stick with it for the moment.